### PR TITLE
Mm/configure nuxt2

### PR DESCRIPTION
**Pull request by MarcMcIntosh**
## Context
#942 
Add a warning about not being able to update the nuxt config file.
Screenshot 2023-04-28 at 14 29 33

## Checklist before requesting a review
- [ ] I hereby declare my code ready for review.
- [ ] If it is a critical feature, I have added tests.
- [ ] The CI is successful.
- [ ] If there could backward compatibility issues, it has been discussed and planned.

### CODE REVIEW

File: packages/adapter-nuxt2/test/plugin-project-init.test.ts [hunk 1]
```diff
@@ -180,3 +180,39 @@ test("Slice Simulator page file is not formatted if formatting is disabled", asy
 		}),
 	);
 });
+
+test("migrates and configures nuxt exporting a function", async (ctx) => {
```
Comment (lihbr) [inline packages/adapter-nuxt2/test/plugin-project-init.test.ts]: I think a better name for the test would be "warns user that Nuxt config could not be updated", not sure this either migrate or configure anything here 🤔

**Review by lihbr — APPROVED** [review]
LGTM, nice job! Just a small one to consider

File: packages/adapter-nuxt2/src/hooks/project-init.ts [hunk 2]
```diff
@@ -38,6 +38,10 @@ const configurePrismicModule = async ({
 }: ConfigurePrismicModuleArgs) => {
 	let nuxtConfigPath = helpers.joinPathFromRoot("nuxt.config.js");
 
+	const endpoint =
```
Comment (hypervillain) [inline packages/adapter-nuxt2/src/hooks/project-init.ts:41]: you can solely rely on `repositoryName` now, it ensures that we add `.cdn` to the endpoint

File: packages/adapter-nuxt2/src/hooks/project-init.ts [hunk 3]
```diff
@@ -47,11 +51,37 @@ const configurePrismicModule = async ({
 		}
 	}
 
-	const mod = await loadFile(nuxtConfigPath);
-	const config =
-		mod.exports.default.$type === "function-call"
-			? mod.exports.default.$args[0]
-			: mod.exports.default;
+	const mod = await loadFile(nuxtConfigPath); // here maybe
```
Comment (hypervillain) [inline packages/adapter-nuxt2/src/hooks/project-init.ts]: not sure what this comment means

File: packages/adapter-nuxt2/src/hooks/project-init.ts [hunk 4]
```diff
@@ -47,11 +51,37 @@ const configurePrismicModule = async ({
...
+
+	let config;
+	try {
+		config =
+			mod.exports.default.$type === "function-call"
+				? mod.exports.default.$args[0]
+				: mod.exports.default;
+	} catch {
+		const errorMessage = `Failed modify ${path.basename(nuxtConfigPath)}`;
```
Comment (hypervillain) [inline packages/adapter-nuxt2/src/hooks/project-init.ts]: @samlfair Could you take a look at this sentence "Failed modify ./nuxt.config.js" and provide something else if needed?

**Review by hypervillain — CHANGES_REQUESTED** [review]

File: packages/adapter-nuxt2/src/hooks/project-init.ts (hunk 2 above)
Comment (lihbr) [reply packages/adapter-nuxt2/src/hooks/project-init.ts:41]: Not with Nuxt 2, it's still running `@prismicio/client@5` and cannot run with just a repository name.

File: packages/adapter-nuxt2/src/hooks/project-init.ts (hunk 3 above)
Comment (lihbr) [reply packages/adapter-nuxt2/src/hooks/project-init.ts]: Nice catch yes 🤔 Let's clean that up

File: packages/adapter-nuxt2/src/hooks/project-init.ts (hunk 4 above)
Comment (lihbr) [reply packages/adapter-nuxt2/src/hooks/project-init.ts]: I think "Failed to update ..." is more common indeed

Comment (hypervillain) [reply packages/adapter-nuxt2/src/hooks/project-init.ts]: yep

File: packages/adapter-nuxt2/src/hooks/project-init.ts (hunk 3 above)
Comment (MarcMcIntosh) [reply packages/adapter-nuxt2/src/hooks/project-init.ts]: I write notes on things to look into when I'm debugging, looks like I left that one in.... oddly enough it was the line below it that was causing the issue

File: packages/adapter-nuxt2/test/plugin-project-init.test.ts (hunk 1 above)
Comment (MarcMcIntosh) [reply packages/adapter-nuxt2/test/plugin-project-init.test.ts]: oh, I forgot to update that one :)

File: packages/adapter-nuxt2/src/hooks/project-init.ts (hunk 4 above)
Comment (MarcMcIntosh) [reply packages/adapter-nuxt2/src/hooks/project-init.ts]: done

Comment (samlfair) [reply packages/adapter-nuxt2/src/hooks/project-init.ts]: I agree with @lihbr. `Failed to update ${...}`  sounds good.

**Review by lihbr — APPROVED** [review]

**Pull request merged**